submissions/ses3332002/friends-app/app.js

```diff
@@ -2,27 +2,33 @@
 const hamburgerMenu = document.querySelector(".hamburger_menu");
 const mainArea = document.querySelector("main");
 const optionForm = document.forms.sorting_n_filtering;
+const asIsOption = document.querySelector("#as_is");
+const allOption = document.querySelector("#all");
+const ageFromOption = document.querySelector("#age_from");
+const ageToOption = document.querySelector("#age_to");
+const searchNameOption = document.querySelector(".filter_name");
+const requestString = "https://randomuser.me/api/?nat=gb,us,fr&results=60&inc=gender,name,location,email,picture,dob";
 let receivedData = [];
 let dataToRender = [];
 let filterCriterionSex = () => true;
 let filterCriterionAge = () => true;
 let filterCriterionName = () => true;
```

Comment:
you don't need this function expression here as a global variables. You only use them in the `changeData` function

Reply:
changeData function is called every time on "input" event on form. But I don't need to reset all the filters and sorting options every time, I need to keep the options that user already set. So I need this options to be stored somewhere outside changeData function, so I need to use global variables.

Reply:
i'm talking about this `filterCriterionSex filterCriterionAge filterCriterionName` vars

Reply:
I'm talking about them too. This vars (and sortCriterion too) contains filter/sorting functions that are to be implemented to array of cards. In case the user choose two ore more filters (like "man", age from 25 to 52, name contains character "a") - I need to keep content of this vars outside of changeData function (where I change this filter/sorting functions depending of user's choice), so they are global vars.
Or maybe I don't understand what you mean.

Reply:
what happens if you move them inside the `changeData` function?

Reply:
For example: 
user set sotring "by age" in form;
changeData function runs, overrides all the filtering and sorting settings, then sets a sorting order by age, prepares array of cards;
render function displays cards sorting by age;
user decides to see only woman sorting by age and set option "women" in form (in addition to already set option to sort by age);
changeData function runs, overrides all the filtering and sorting settings, then sets a filter to show only women, prepares array of cards;
render function displays cards filtered by sex but without sorting by age (because this setting was reassigned in changeData function when it worked last time);
user says "What the hell!?"
I can make a demo if you want to see this version.